bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/template/TemplateRegistryImpl.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core.template;
+
+import java.util.Collection;
+import java.util.Locale;
+
+import org.eclipse.smarthome.automation.template.Template;
+import org.eclipse.smarthome.automation.template.TemplateRegistry;
+import org.eclipse.smarthome.core.common.registry.AbstractRegistry;
+
+/**
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public class TemplateRegistryImpl extends AbstractRegistry<Template, String> implements TemplateRegistry {
```

Comment:
This class should actually not extend AbstractRegistry since it does not offer add, update and remove of elements.